ports/stm/common-hal/os/__init__.c

```diff
@@ -77,11 +77,10 @@ bool common_hal_os_urandom(uint8_t *buffer, uint32_t length) {
         uint32_t start = HAL_GetTick();
         //the HAL function has a timeout, but it isn't long enough, and isn't adjustable
         while(!(__HAL_RNG_GET_FLAG(&handle,RNG_FLAG_DRDY)) && ((HAL_GetTick() - start) < RNG_TIMEOUT));
-        //
         if (HAL_RNG_GenerateRandomNumber(&handle, &temp) != HAL_OK) {
             mp_raise_ValueError(translate("Random number generation error"));
         }
-        *buffer = (uint8_t)temp;
+        buffer[i] = (uint8_t)temp;
```

Comment:
Want to do a for loop here too? It will likely speed this up because it'll use all 32 bits from the lower level instead of just the first 8.

Reply:
I didn't even bother to check that that function returns a 32bit number, good point. I'll swap it over to match.

Reply:
If speed is a concern, do you want me to add in some way of keeping the TRNG on rather than enabling and disabling it every time? Or is that NBD?

Reply:
I wouldn't worry about turning it off and on. I just know that generating the numbers can take time too so you might as well use all of the bits.

Reply:
Ok, I've modified it to match the other two ports.